imgparse/parser.py

```diff
@@ -0,0 +1,563 @@
+                    float(self.xmp_data[self.xmp_tags.WAVELENGTH_CENTRAL])
+                ]
+                wavelength_fwhm = [float(self.xmp_data[self.xmp_tags.WAVELENGTH_FWHM])]
+
+            return central_wavelength, wavelength_fwhm
+        except KeyError:
+            raise ParsingError(
+                "Couldn't parse wavelength data. Sensor might not be supported"
+            )
+
+    def bandnames(self) -> list[str]:
+        """Get the name of each band of an image."""
+        try:
+            try:
+                return parse_seq(self.xmp_data[self.xmp_tags.BANDNAME])
+            except TypeError:
+                return [self.xmp_data[self.xmp_tags.BANDNAME]]
+        except KeyError:
+            raise ParsingError(
+                "Couldn't parse bandnames. Sensor might not be supported"
+            )
+
+    def lens_model(self) -> str:
+        """Get the lens model of an image from a Sentera Camera."""
+        try:
+            return (
+                str(self.exif_data["Image LensModel"].values)
+                if self.exif_data.get("Image LensModel")
+                # will return KeyError if both are not found
+                else str(self.exif_data["EXIF LensModel"].values)
+            )
+
+        except KeyError:
+            raise ParsingError(
+                "Couldn't parse lens model. Sensor might not be supported"
+            )
+
+    def serial_number(self) -> int:
+        """
+        Get the serial number of the sensor.
+
+        Expects serial number to be parsable as an integer.
+        """
+        try:
+            return int(self.exif_data["Image BodySerialNumber"].values)
+        except (KeyError, ValueError):
+            raise ParsingError(
+                "Couldn't parse sensor version. Sensor might not be supported"
+            )
+
+    def irradiance(self) -> float:
+        """Get the Irradiance value of an image captured by a sensor with an DLS module."""
+        try:
+            return float(self.xmp_data[self.xmp_tags.IRRADIANCE])
+        except KeyError:
+            raise ParsingError(
+                "Couldn't parse irradiance value. Sensor might not be supported"
+            )
+
+    def capture_id(self) -> str:
+        """
+        Get unique id for a single capture event.
+
+        This unique id is consistent across bands for multispectral cameras.
+        """
+        try:
+            if self.make() == "Sentera":
+                try:
+                    # Firmware version >=2.1.0
+                    return f"{self.xmp_data['Camera:FlightUUID']}_{self.xmp_data['Camera:CaptureUUID']}"
+                except KeyError:
+                    # Firmware version <2.1.0
+                    return f"{self.xmp_data['Camera:FlightUniqueID']}_{self.xmp_data['Camera:ImageUniqueID']}"
+            else:
+                return str(self.xmp_data[self.xmp_tags.CAPTURE_UUID])
+        except KeyError:
+            raise ParsingError("Couldn't determine unique id")
```

Comment:
Previously the image number was extracted from the filename as a fallback via:

    re.findall(r"IMG_(\d+)", os.path.basename(image_path))[0]
    
However with @joeyfranck56's changes, we support more than just Sentera sensors so this won't work. @SamCWill how often do we hit this fallback case and do you have any ideas how to make it more generic?

Reply:
@elliott-imhoff I feel like it won't be very often that 6X images don't have the UUID tags.  I do think it is not a good idea to use the old fallback as images can have the same number and the desktop app renames images in a random method.  I think you have the best solution going forward and older datasets may need to use an older version.